src/node_http2.cc

```diff
@@ -2372,8 +2372,7 @@ int Http2Stream::DoWrite(WriteWrap* req_wrap,
   CHECK_NULL(send_handle);
   Http2Scope h2scope(this);
   if (!is_writable() || is_destroyed()) {
-    req_wrap->Done(UV_EOF);
```

Comment:
@RafaelGSS After investigation, I find out that this `if` statement should be turned to an assertion. Because the JS code make sure that whenever we do a write on a stream, the stream is writable and not destroyed.

Reply:
Changing it to an assertion (`CHECK`) will crash the application whenever someone tries to write in a destroyed stream, is that expected?

Reply:
no.

Reply:
JS code in /lib will not let this happens. Without misusing the internals, the case that captured by this `if` can not happen. If you insist keeping this `if`, then apparently, this function should return an error code immediately and never invoke the cb, because this is what every asynchronous API should do.

Reply:
If that happens when misusing the internals, adding the `CHECK` make sense to me.